core/trino-main/src/main/java/io/trino/cost/PlanNodeStatsEstimateMath.java

```diff
@@ -135,6 +144,82 @@ public static PlanNodeStatsEstimate capStats(PlanNodeStatsEstimate stats, PlanNo
+                            .map(SymbolStatsEstimate::getNullsFraction)
+                            .reduce(MoreMath::minExcludeNaN)
+                            .orElseThrow();
+
+                    double averageRowSize = symbolStatsEstimates.stream()
+                            .map(SymbolStatsEstimate::getAverageRowSize)
+                            .reduce(MoreMath::averageExcludingNaNs)
+                            .orElseThrow();
+
+                    result.addSymbolStatistics(symbol, SymbolStatsEstimate.builder()
+                            .setStatisticsRange(intersect)
+                            .setNullsFraction(nullsFraction)
+                            .setAverageRowSize(averageRowSize)
+                            .build());
+                });
+        return result.build().getSymbolStatistics();
+    }
+
+    public static double estimateCorrelatedConjunctionRowCount(
+            PlanNodeStatsEstimate input,
+            List<PlanNodeStatsEstimate> estimates,
+            double independenceFactor)
+    {
+        checkArgument(!estimates.isEmpty(), "estimates is empty");
+        if (input.isOutputRowCountUnknown() || input.getOutputRowCount() == 0) {
+            return input.getOutputRowCount();
+        }
+        List<PlanNodeStatsEstimate> knownSortedEstimates = estimates.stream()
+                .filter(estimateInfo -> !estimateInfo.isOutputRowCountUnknown())
+                .sorted(comparingDouble(PlanNodeStatsEstimate::getOutputRowCount))
+                .collect(toImmutableList());
+        if (knownSortedEstimates.isEmpty()) {
+            return NaN;
+        }
+
+        PlanNodeStatsEstimate combinedEstimate = knownSortedEstimates.get(0);
+        double combinedSelectivity = combinedEstimate.getOutputRowCount() / input.getOutputRowCount();
+        double combinedIndependenceFactor = 1.0;
+        // For independenceFactor = 0.75 and terms t1, t2, t3
+        // Combined selectivity = (t1 selectivity) * ((t2 selectivity) ^ 0.75) * ((t3 selectivity) ^ (0.75 * 0.75))
+        // independenceFactor = 1 implies the terms are assumed to have no correlation and their selectivities are multiplied without scaling.
+        // independenceFactor = 0 implies the terms are assumed to be fully correlated and only the most selective term drives the selectivity.
+        for (int i = 1; i < knownSortedEstimates.size(); i++) {
+            PlanNodeStatsEstimate term = knownSortedEstimates.get(i);
+            combinedIndependenceFactor *= independenceFactor;
+            combinedSelectivity *= Math.pow(term.getOutputRowCount() / input.getOutputRowCount(), combinedIndependenceFactor);
+        }
+        double outputRowCount = input.getOutputRowCount() * combinedSelectivity;
+        boolean hasUnestimatedTerm = estimates.stream().anyMatch(PlanNodeStatsEstimate::isOutputRowCountUnknown);
+        return hasUnestimatedTerm ? outputRowCount * UNKNOWN_FILTER_COEFFICIENT : outputRowCount;
```

Comment:
add `TODO` about using `UNKNOWN_FILTER_COEFFICIENT` only when `default-filter-factor=true`

Reply:
This is unrelated to current PR

Reply:
Still, `TODO` here would be great (since this is new code actually) to not forget about it (and places that need to be changed)